stencils/pace/stencils/testing/grid.py

```diff
@@ -1089,3 +1089,93 @@ def _grid_indexing_axis_offsets(
     domain: Tuple[int, ...],
 ) -> Mapping[str, gtscript.AxisIndex]:
     return grid.axis_offsets(origin=origin, domain=domain)
+
+
+@dataclasses.dataclass(frozen=True)
+class DriverGridData:
+    """
+    Terms used to Apply Physics changes to the Dycore.
+    """
+
+    vlon1: FloatFieldIJ
+    vlon2: FloatFieldIJ
+    vlon3: FloatFieldIJ
+    vlat1: FloatFieldIJ
+    vlat2: FloatFieldIJ
+    vlat3: FloatFieldIJ
+    edge_vect_w: FloatFieldI
+    edge_vect_e: FloatFieldI
+    edge_vect_s: FloatFieldIJ
+    edge_vect_n: FloatFieldIJ
+    es1_1: FloatFieldIJ
+    es1_2: FloatFieldIJ
+    es1_3: FloatFieldIJ
+    ew2_1: FloatFieldIJ
+    ew2_2: FloatFieldIJ
+    ew2_3: FloatFieldIJ
```

Comment:
Thanks for updating!

Ideally we don't - the way to avoid duplicating it is to have sub-containers as attributes on higher containers, so that the higher containers can say "I have this sub-container attribute" and make use of its documentation. This is what we do in our init calls - the init signatures say "I want this container", and you'd look at the container's documentation for variable descriptions.

When an attribute is actually needed on multiple containers to be read directly on that container, we need to document each use.